app/src/main/java/org/mozilla/focus/telemetry/TelemetrySettingsProvider.kt

```diff
@@ -0,0 +1,37 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.telemetry
+
+import android.content.Context
+import org.mozilla.focus.tiles.BundledTilesManager
+import org.mozilla.focus.tiles.CustomTilesManager
+import org.mozilla.telemetry.measurement.SettingsMeasurement
+
+/**
+ * A SettingsProvider that provides custom value getters when settings values
+ * can't just be retrieved from SharedPreferences.
+ */
+class TelemetrySettingsProvider(private val appContext: Context) : SettingsMeasurement.SharedPreferenceSettingsProvider() {
+
+    override fun containsKey(key: String?): Boolean {
+        return key == PREF_CUSTOM_HOME_TILE_COUNT ||
+                key == PREF_TOTAL_HOME_TILE_COUNT ||
+                super.containsKey(key)
+    }
+
+    override fun getValue(key: String?): Any? = when (key) {
+        PREF_CUSTOM_HOME_TILE_COUNT -> CustomTilesManager.getInstance(appContext).tileCount
+        PREF_TOTAL_HOME_TILE_COUNT -> CustomTilesManager.getInstance(appContext).tileCount +
+                BundledTilesManager.getInstance(appContext).tileCount
```

Comment:
I think here you might need to remove the home tiles blacklist.

Reply:
The bundled tiles cache removes any tiles on the blacklist so we can use this value directly.